tests/Aspire.Hosting.Tests/ResourceExtensionsTests.cs

```diff
@@ -25,4 +25,102 @@ public void TryGetContainerImageNameReturnsCorrectFormatWhenShaNotSupplied()
         Assert.True(container.Resource.TryGetContainerImageName(out var imageName));
         Assert.Equal("grafana/grafana:10.3.1", imageName);
     }
+
+    [Fact]
+    public async Task GetEnvironmentVariableValuesAsyncReturnCorrectVariablesInRunMode()
+    {
+        var builder = DistributedApplication.CreateBuilder();
+        var container = builder.AddContainer("elasticsearch", "library/elasticsearch", "8.14.0")
+         .WithEnvironment("discovery.type", "single-node")
+         .WithEnvironment("xpack.security.enabled", "true")
+         .WithEnvironment(context =>
+         {
+             context.EnvironmentVariables["ELASTIC_PASSWORD"] = "123456";
```

Comment:
To avoid credscan failure:
```suggestion
             context.EnvironmentVariables["ELASTIC_PASSWORD"] = "p@ssw0rd1";
```